src/block-tracker-inspector.ts

```diff
@@ -13,37 +16,64 @@ interface BlockTrackerInspectorMiddlewareOptions {
   blockTracker: PollingBlockTracker;
 }
 
+function hasProperty<Property extends string>(
+  object: unknown,
+  property: Property,
+): object is Record<Property, unknown> {
+  return Object.hasOwnProperty.call(object, property);
+}
+
+function getResultBlockNumber(
+  response: PendingJsonRpcResponse<unknown>,
+): string | undefined {
+  const { result } = response;
+  if (
+    !result ||
+    typeof result !== 'object' ||
+    !hasProperty(result, 'blockNumber')
+  ) {
+    return undefined;
+  }
+
+  if (result.blockNumber && typeof result.blockNumber === 'string') {
```

Comment:
```suggestion
  if (typeof result.blockNumber === 'string') {
```

Redundant since we already check for `result` just before?

Reply:
Good catch - it looks like this can be removed

Reply:
Resolved here: 5ebedbf7a144f78a30749198d3441c0ae41afa48